pkg/webhooks/utils/block.go

```diff
@@ -43,8 +43,16 @@ func GetBlockedMessages(engineResponses []engineapi.EngineResponse) string {
 	for _, er := range engineResponses {
 		ruleToReason := make(map[string]string)
 		for _, rule := range er.PolicyResponse.Rules {
-			if rule.Status() != engineapi.RuleStatusPass {
-				ruleToReason[rule.Name()] = rule.Message()
+			if len(engineResponses) > 1 {
```

Comment:
Why do we need this check?

Reply:
If we have policies more than one , then we can skip the denial message of preconditions. 
But if only one policy is available , then it should not

Reply:
For each policy, there is a corresponding engine response. And, for each engine response, there is a policy response that contains multiple rule responses if the policy has multiple rules. So, we are simply looping through the engine responses and for each, we loop through the rule responses and for each, we should check if the rule status is equal to `skip` or `pass`, then we don't add its message in `ruleToReason`. Hence, there is no need to perform this check.